src/Stratis.Bitcoin.Features.Miner/BlockProvider.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using NBitcoin;
+using Stratis.Bitcoin.Mining;
+
+namespace Stratis.Bitcoin.Features.Miner
+{
+    public sealed class BlockProvider : IBlockProvider
+    {
+        private readonly PowBlockDefinition powBlockDefinition;
+        private readonly PosBlockDefinition posBlockDefinition;
+
+        /// <summary>
+        /// This is constructor is called by dependency injection.
```

Comment:
Isn't this assumed?  Do you need to state it's called by DI?

Reply:
Fixed